Product definition updates for C3

jeremyh opened the pull request:
(see commit messages below: they're clean, so no point repeating here?)

The only potentially-breaking change is increasing the ODC requirement to version 1.8, as its needed for license fields in products. But all DEA environments should now use 1.8, I assume?

jeremyh committed:
Include license in all eo3 products

jeremyh committed:
Add C3 L1 GA products

jeremyh committed:
Add landsat_scene_id field to ARD

Needed by Duncan for scene filtering.

jeremyh committed:
Add landsat_data_type field for L1

Requested by Simon for checking L1T/etc levels within the product.

jeremyh committed:
Update L1 GA definitions for new conventions

To match recent updates by @simonaoliver

jeremyh committed:
Raise ODC requirement to 1.8.0

Needed for license fields in products... and should be be used by
everyone regardless.

jeremyh commented:
@simonaoliver: Are these licenses correct for our Landsat products?  `license: CC-BY-4.0` (ie: https://creativecommons.org/licenses/by/4.0/ )

simonaoliver commented:
Yes - I believe CC-BY 4.0 applies here @jeremyh .

simonaoliver reviewed (approved):
description now agrees with convention

jeremyh committed:
Make Travis install ODC through pypi

If we want to test it against dev versions of ODC, we
could create a separate optional runner for that.

Currently, installing through the package repo breaks the
build here too often, and everyone's too busy to fix it. The
breakages haven't been due to ODC bugs, only due to packaging
oddities.

jeremyh committed:
Update C2 metadata-types to match actual NCI index

jeremyh committed:
Make tsmask product match NCI db

It seems to only been added as an 'eo' type.

jeremyh committed:
Clean-up yaml warnings

jeremyh committed:
Add fixed fields to eo3 product metadata

jeremyh committed:
Include all landsat collection fields

jeremyh committed:
Bump setuptools, pip versions

There's a warning in the Travis log.

jeremyh committed:
Remove eo_plus from expected types

jeremyh committed:
Add ga l1 products to expected products
